pkg/driver/driver_aws.go

```diff
@@ -127,16 +134,17 @@ func (d *AWSDriver) Create() (string, string, error) {
 	}
 	metrics.APIRequestCount.With(prometheus.Labels{"provider": "aws", "service": "ecs"}).Inc()
 
-	return d.encodeMachineID(d.AWSMachineClass.Spec.Region, *runResult.Instances[0].InstanceId), *runResult.Instances[0].PrivateDnsName, nil
-}
-
-func (d *AWSDriver) generateSecurityGroups(sgIDs []string) ([]*string, error) {
-
-	var securityGroupIDs []*string
-	for _, sg := range sgIDs {
-		securityGroupIDs = append(securityGroupIDs, aws.String(sg))
+	_, err = svc.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
+		InstanceId:      runResult.Instances[0].InstanceId,
+		SourceDestCheck: &ec2.AttributeBooleanValue{Value: d.AWSMachineClass.Spec.SourceDestCheck},
```

Comment:
Same comment as above, can we please avoid executing this code-path if `SourceDestCheck` is not set in the MachineClass. 
- It could be, we exeute `ModifyInstanceAttribute` only if the `SourceDeskCheck` is set.

Reply:
And there we've got some peculiar design decision on the AWS go-sdk team's part. Namely, the `AttributeBooleanValue` thing. To skip e2e testing and possible breaking I've guarded it with the `if type !=nil ` idiom.

Reply:
Sure, thanks for the change, it looks good now.